Update transactions.md (#321)
JoshOrndorff/recipes: text/3-entrees/off-chain-workers/transactions.md

[ocw-runtime](https://github.com/substrate-developer-hub/recipes/tree/master/runtimes/ocw-runtime).
That runtime can be used in the kitchen node.

In order to utilize the Offchain worker, the node must inject keys into its keystore. This ensures that it is

> enabled with a feature flag, however in order to do this we need to enable the `ocw-runtime` by editing `nodes/kitchen-node/Cargo.toml`. 

Reviewer: Addie, let me rewrite these two lines and you check if that's better.

[ocw-runtime](https://github.com/substrate-developer-hub/recipes/tree/master/runtimes/ocw-runtime).
That runtime can be used in the kitchen node.

In order to utilize the off-chain worker, the node must inject keys into its keystore. To do so, we
open the `nodes/kitchen-node/Cargo.toml` file, enable the `ocw-runtime` package and comment out the
`super-runtime` package.

> Then we build the kitchen node with `ocw` feature flag:

Reviewer: @wheresaddie I have updated these lines, see if you are good with them.

## Life-cycle of Off-chain Worker

> Running the `kitchen-node`we see off-chain worker runs after each block generation

Reviewer: ```suggestion
Running the `kitchen-node` we see off-chain worker runs after each block generation
```

...
```

If we refer to the code at
[`pallets/offchain-demo/src/lib.rs`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/offchain-demo/src/lib.rs),
there is an `offchain_worker` function inside `decl_module!`. This is the entry point of the
off-chain worker which is executed once per block generation, so all the off-chain
logic is here.

Two kinds of transactions can be sent back on-chain from off-chain workers, **Signed Transactions**
and **Unsigned Transactions**. Signed transactions are used if the transaction requires the sender

> to be specified. Unsigned transactions are used when the sender does not need to be known and
> additional logic in the code provides additional data verification. 

Reviewer: ```suggestion
to be specified. Unsigned transactions are used when the sender does not need to be known, but this would require additional logic be in place to provide extra data verification. 
```

> Let's walk through how to set up each one:

Reviewer: ```suggestion
Let's walk through how to set each one up:
```

## Signed Transactions

> **Notes**: This example will have account `Alice` submitting signed transactions to the node in
off-chain worker, and the transactions reflecting the associated fees. If we run the node in development
mode (with `--dev` flag) using the default sr25519 crypto signature, `Alice` will have initial funds
deposited and this example will run all fine.
>
> If this is not done in development mode or switched to another crypto signature, please
> be aware that `Alice` must be setup and be funded to run this example.

### Setup

> For signed transactions, we have to define another sub-module src:

Reviewer: ```suggestion
For signed transactions, we have to define a crypto signature sub-module:
```

}
```

> This is the application key is a prefix for the pallet in underlying storage.

Reviewer: ```suggestion
`KEY_TYPE` is the application key prefix for the pallet in underlying storage.
```

}
```

> This associated type needs to be specified by the runtime
> (implement this pallet trait).

Reviewer: ```suggestion
This associated type needs to be specified by the runtime when it implements this pallet trait.
```

Now if we build the `kitchen-node`, we will see the compiler return with three trait
bounds that are not satisfied: `Runtime: frame_system::offchain::CreateSignedTransaction`,
`frame_system::offchain::SigningTypes`, and `frame_system::offchain::SendTransactionTypes`. 

> We learn that when using `SubmitSignedTransaction`, we also need to have our runtime
> implement 

Reviewer: ```suggestion
We also learn that when using `SubmitSignedTransaction`, the runtime need to implement 
```

[`CreateSignedTransaction` trait](https://substrate.dev/rustdocs/v2.0.0-rc5/frame_system/offchain/trait.CreateSignedTransaction.html).

> So this needs to be implemented to our runtime.

Reviewer: ```suggestion
```

are going to submit back to on-chain, and passing it to
`T::SubmitSignedTransaction::submit_signed(call)`.

> Notice that we run a loop in the returned result. This implies that a call may make

Reviewer: ```suggestion
Notice that we run a loop in the returned result. This implies that the call may make
```

multiple transactions and return multiple results. The call is both signing and sending
the transaction with each of the accounts that can be found locally under the application crypto
(which we defined earlier in `pub mod crypto {...}`). This can be seen as the local accounts that
are managed under this pallet namespace. As we only have one key in the app crypto, so only
one signed transaction is made.

Eventually, the `call` transaction will be made on-chain via the `create_transaction` function we defined
earlier when we implemented `CreateTransaction` trait in our runtime.

> The local account is inserted in the pallet app crypto, and lives in

Reviewer: ```suggestion
The local account used to sign the transaction is inserted in the pallet app crypto, and lives in
```

}
```

> By implementing `ValidateUnsigned`, and the allowance logic is added inside the `validate_unsigned`
> function. We can verify if the call is to `Call::submit_number_unsigned` and returns `Ok()` otherwise, `InvalidTransaction::Call`.

Reviewer: ```suggestion
By implementing `ValidateUnsigned`, the allowance logic is added inside the `validate_unsigned`
function. We verify that if the call is `Call::submit_number_unsigned` we return `Ok()`, otherwise `InvalidTransaction::Call`.
```
